Modules/_hashopenssl.c

```diff
@@ -1096,12 +1098,52 @@ generate_hash_name_list(void)
     return state.set;
 }
 
+/* LibreSSL doesn't support FIPS:
+   https://marc.info/?l=openbsd-misc&m=139819485423701&w=2
+
+   Ted Unangst wrote: "I figured I should mention our current libressl policy
+   wrt FIPS mode.  It's gone and it's not coming back." */
+#ifndef LIBRESSL_VERSION_NUMBER
+/*[clinic input]
+_hashlib.get_fips_mode -> int
+
+Determine the OpenSSL FIPS mode of operation.
+
+Effectively any non-zero return value indicates FIPS mode;
+values other than 1 may have additional significance.
+
+See OpenSSL documentation for the FIPS_mode() function for details.
+[clinic start generated code]*/
+
+static int
+_hashlib_get_fips_mode_impl(PyObject *module)
+/*[clinic end generated code: output=87eece1bab4d3fa9 input=c2799c3132a36d6c]*/
+
+{
+    int result = FIPS_mode();
```

Comment:
If you perform an OpenSSL ERR stack check, then you must clear the OpenSSL ERR stack before you call the function.

Reply:
Alright, I added a call to ERR_clear_error() at the function entry point.